### ISSUE TITLE
[Question from Etsy] How much time do you think you'll need to plan and execute your migration?

### ISSUE THREAD

**Issue by mikabro** [etsy/open-api#44]
Developers,

It's Etsy's Open API/Integrations team! We want feedback on some specific topics, so we're posting questions and would love to get your opinions. 

How much time do you believe you'll need to plan and execute your migration? In your reply, it would also be helpful to hear about any constraints you have and what factors influence the length of time needed.

**Comment by turboturtleD** [comment]
Aside from how long, I think WHEN is entirely more critical.  Sunsetting in Q4 is not good timing for an e-commerce platform.  This is the busiest time of year on Etsy and those who rely on integrations (in-house or 3rd party) will struggle to deal with migration/migration-related-bugs while trying to simultaneously dealing with the heaviest traffic of the year.  I would put a BIG OLD PLEASE out there to not sunset v2 in Q4.

**Comment by benjymouse** [comment]
If possible, please don't sunset v2 API until end of Q1 next year. 

We plan to go into code freeze about 4 weeks before Black Friday, so if you need to close v2 API in Q4, please give us plenty of heads up.

**Comment by kievins** [comment]
I agree with the above comments about concerns regarding the Q4 sunset. 

Another concern is that (as far as I know) we would have to ask all users to re-connect their app to get the new API tokens. In our experience this causes a lot of issues for end users. Many (less active) users will only notice this when they run into issues due to data not flowing between Etsy and apps. Ideally there would be a way to save them the trouble and convert the old API tokens into new ones.

Migrating the code should not take more than a month or two, but ironing out any new issues that might arise will likely take about as long if not more.

**Comment by missnicbic** [comment]
Where has the Q4 sunset date been discussed? Switching our app over to v3 would be an extremely large undertaking for us and as mentioned having this occur in one of the busiest times of the year for Etsy sellers would be disastrous.
